### DISCUSSION TITLE
OpenAIPromptExecutionSettings.ResponseFormat should be using for structed outputs?

### DISCUSSION THREAD

**Discussion by ConnorGuard** [microsoft/semantic-kernel#9071] [Q&A] [answered]
Hi,

I was reading this recently published article: https://devblogs.microsoft.com/semantic-kernel/using-json-schema-for-structured-output-in-net-for-openai-models/

You were using **OpenAIPromptExecutionSettings** to use the property **ResponseFormat**

When I use it, I get the message: 'Feature is for evaluation purposes only and is subject to change or removal in future updates.'

It also seems that **PromptExecutionSettings** doesnt have this property.

Is **ResponseFormat** on **OpenAIPromptExecutionSettings** the correct method I should be using for structed outputs? Will this being changed or removed soon?

Thanks :)

**Comment by moonbox3** [comment, marked as answer]
Hi @ConnorGuard, thanks for your question. Since the json_schema structured outputs is OpenAI specific functionality, yes, OpenAIPromptExecutionSettings is correct. 

The reason you're seeing the `Feature is for evaluation purposes...` message is because this is experimental. It's still experimental for OpenAI, and thus subject to change on their side, which forces changes on our side.

**Reply by hootanht** [reply]
Thanks for your response, @moonbox3! I appreciate the clarification. Given that this functionality is experimental, is there any recommended workaround or method to use `OpenAIPromptExecutionSettings.ResponseFormat` for structured outputs right now? Or should I wait for a more stable release from OpenAI before proceeding with its integration?

**Comment by BenCoden** [comment]
@moonbox3  So, does `ResponseFormat` only work with complex types? For example, I can't say `OpenAIPromptExecutionSettings.ResponseFormat = "xaml"`

**Reply by moonbox3** [reply]
They don’t have to be complex, but they need to be types. We generate a JSON schema from them, and pass them to the model, so it can use it to format the returned response.
